packages/astro/src/runtime/server/jsx.ts

```diff
@@ -91,7 +91,11 @@ Did you forget to import the component or is it possible there is a typo?`);
 						props[key] = value;
 					}
 				}
-				const html = markHTMLString(await renderToString(result, vnode.type as any, props, slots));
+				const str = await renderToString(result, vnode.type as any, props, slots);
+				if (str instanceof Response) {
+					throw str;
+				}
```

Comment:
This is not ideal, but I'm following the API we had before:

https://github.com/withastro/astro/blob/d8bae784b4a3b555c213febe6334373d89c7e089/packages/astro/src/runtime/server/render/astro/factory.ts#L32-L35